packages/react/src/use-chat.ts

```diff
@@ -513,12 +516,58 @@ By default, it's set to 0, which will disable the feature.
 
       event?.preventDefault?.();
 
+      let parts: ContentPart[] = [];
+
+      if (input) {
+        parts.push({
+          type: 'text',
+          text: input,
+        });
+      }
+
+      if (options.files) {
+        for (const file of Array.from(options.files)) {
+          const { type } = file;
+
+          if (type.startsWith('image/')) {
+            const base64 = await new Promise<string>((resolve, reject) => {
+              const reader = new FileReader();
+              reader.onload = readerEvent => {
+                resolve(readerEvent.target?.result as string);
+              };
+              reader.onerror = error => reject(error);
+              reader.readAsDataURL(file);
+            });
+
+            parts.push({
+              type: 'image',
+              image: base64,
+            });
+          } else if (type.startsWith('text/')) {
```

Comment:
how would text resources be handled on the backend? can we just send all resources that are not images as generic "file" with base64 encoding?

Reply:
also maybe rewrite both using async/await and see if you can refactor to remove duplication

Reply:
ah i see we are sending it in the parts that need to be compatible. hmm. i'm wondering if we want to be more flexible, since this is for client server communication, not server llm communication. someone might eg want to send a pdf, that is then chunked on the server and used for RAG in that request

Reply:
don't this is relevant anymore sine we updated `parts` to `attachments` and added support for urls for server side processing